pkg/logging/constants.go

```diff
@@ -92,3 +94,31 @@ func FromPrefix(prefix string, obj Object) []interface{} {
 		gvk.GroupVersion(),
 	}
 }
+
+// WithUser adds user identifiers to the logger.
+func WithUser(logger logr.Logger, u user.Info) logr.Logger {
+	return logger.WithValues(FromUser(u)...)
+}
+
+// FromUser provides the structured logging fields that identify a user, prefixing with 'user'.
+func FromUser(u user.Info) []interface{} {
+	return FromUserPrefix(u, "user")
+}
+
+// FromUserPrefix provides the structured logging fields that identify a user, allowing any prefix.
+func FromUserPrefix(u user.Info, prefix string) []interface{} {
+	return []interface{}{
+		prefix + ".name", u.GetName(),
+		prefix + ".uid", u.GetUID(),
```

Comment:
I think that `UID`s can get very long, maybe we could just log some portion of it?